armi/materials/material.py

```diff
@@ -32,6 +33,36 @@
 FAIL_ON_RANGE = False
 
 
+def parentAwareDensityRedirect(f):
+    """Wrap Material.density to warn people about potential problems.
+
+    If a Material is linked to a Component, ``Material.density`` may produce
+    different results from ``Component.density``. The component's density
+    is considered the source of truth because it incorporates changes in volume,
+    composition, and temperature in concert with the state of the reactor.
+
+    See Also
+    --------
+    - https://github.com/terrapower/armi/issues/1440
+
```

Comment:
```suggestion
```

Reply:
We don't link to PRs or Issues in the code.

The idea is that either that information is important enough to write in a docstring (or the docs), or it's not.

Reply:
Fair enough